cgi-bin/DW/Logic/ProfilePage.pm

```diff
@@ -765,7 +768,17 @@ sub external_services {
     my $remote = $self->{remote};
     my @ret;
 
-    return () unless $u->is_personal && ( $u->share_contactinfo($remote) || $self->{viewall} );
+    return [] unless $u->is_personal && ( $u->share_contactinfo($remote) || $self->{viewall} );
+
+    # this has gotten big enough that we should improve database efficiency
```

Comment:
oh!! I always wondered what that method was about -- so I guess it lets you do this as a batch request without having to _remember_ that it's a batch request elsewhere in the code. TIL.

Reply:
Yeah, it takes all the listed props and batches them into a single DB lookup, and the subsequent prop calls get the info from the hashref or from memcache.